api/lib/infrastructure/repositories/campaign-participation-repository.js

```diff
@@ -63,12 +65,21 @@ module.exports = {
   },
 
   async save(campaignParticipation, domainTransaction = DomainTransaction.emptyTransaction()) {
-    const knexConn = domainTransaction.knexTransaction || knex;
-    const [newCampaignParticipation] = await knexConn('campaign-participations')
-      .insert(_adaptModelToDb(campaignParticipation))
-      .returning('*');
-
-    return new CampaignParticipation(newCampaignParticipation);
+    try {
+      const knexConn = domainTransaction.knexTransaction || knex;
+      const [newCampaignParticipation] = await knexConn('campaign-participations')
+        .insert(_adaptModelToDb(campaignParticipation))
+        .returning('*');
+
+      return new CampaignParticipation(newCampaignParticipation);
+    } catch (error) {
+      if (knexUtils.isUniqConstraintViolated(error)) {
+        const { userId, campaignId } = campaignParticipation;
+        throw new AlreadyExistingEntityError(
+          `A campaignParticipation already exists for the user ${userId} and campaign ${campaignId}.`
+        );
+      }
```

Comment:
ce que suggère @alexandrecoin :
```suggestion
    } catch (error) {
      if (knexUtils.isUniqConstraintViolated(error)) {
        const { userId, campaignId } = campaignParticipation;
        throw new AlreadyExistingEntityError(
          `A campaignParticipation already exists for the user ${userId} and campaign ${campaignId}.`
        );
      }
      throw error;
```

Reply:
Fait dans https://github.com/1024pix/pix/pull/3756#discussion_r754307317